File: backend/src/main/java/com/burndown/entity/Project.java
package com.burndown.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "projects")
@EntityListeners(AuditingEntityListener.class)
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "project_key", nullable = false, unique = true, length = 20)
    private String projectKey;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ProjectType type;

    @Column(length = 20)
    private String visibility = "PRIVATE";

    @Column(name = "start_date")
    private java.time.LocalDate startDate;

    @Column(name = "end_date")
    private java.time.LocalDate endDate;

    @Column(name = "owner_id", nullable = false)
    private Long ownerId;

    @Column(length = 20)
    private String status = "ACTIVE";

    @Column(columnDefinition = "jsonb")
    @org.hibernate.annotations.JdbcTypeCode(org.hibernate.type.SqlTypes.JSON)
    private String settings = "{}";

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum ProjectType {
        SCRUM, KANBAN
    }
}

File: frontend/src/api/project.ts
import api from './axios'

export interface Project {
  id: number
  name: string
  description: string
  projectKey: string
  type: string
  ownerId: number
  createdAt: string
}

export interface CreateProjectRequest {
  name: string
  description?: string
  projectKey: string
  type: string
  visibility?: string
  startDate?: string
  endDate?: string
}

export const projectApi = {
  getAll: () => api.get<Project[]>('/projects'),
  getById: (id: number) => api.get<Project>(`/projects/${id}`),
  getMyProjects: () => api.get<Project[]>('/projects/my-projects'),
  create: (data: CreateProjectRequest) => api.post<Project>('/projects', data),
}

export const getProjects = async () => {
  const response = await projectApi.getAll()
  return response.data
}

export const createProject = async (data: CreateProjectRequest) => {
  const response = await projectApi.create(data)
  return response.data
}
